timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -187,7 +187,7 @@ void __global__ k_scatter_accum(
         return;
     }
 
-    array[unique_idxs[idx] * stride + stride_idx] += gathered_array[idx * stride + stride_idx];
+    atomicAdd(array + (unique_idxs[idx] * stride + stride_idx), gathered_array[idx * stride + stride_idx]);
```

Comment:
All potentials need to do atomic adds to the buffers, which would make this a more fragile test. Lots of tests fail without this change.

Reply:
Have you tried running this with the racecheck tool yet? https://docs.nvidia.com/compute-sanitizer/ComputeSanitizer/index.html#racecheck-tool

(not sure if it'd be able to detect this specific issue, but seems like a good idea when introducing parallelism)

Reply:
Good question, I hadn't tried. Will give it a shot. I imagine it would pick it up

Reply:
Using `compute-sanitizer --tool racecheck` it did not catch the du_dp bug I ran into. Think the profiling changes the runtime enough to not hit the parallel write, as I could get the test to pass 2/3 of the time locally.

Going through the rest of the potentials don't see any non-atomic updates left.